vuecomponents/tool/src/main/frontend/src/components/dynamic-rubric.vue

```diff
@@ -0,0 +1,338 @@
+      const resp1 = fetch(associationUrl, {
+        credentials: "include",
+        headers: { "Content-Type": "application/json" },
+      }).then(r => {
+        if (r.ok) {
+          return r.json();
+        }
+
+        throw new Error("Network error while getting association");
+        }).then(association => {
+          this.association = association;
+          //if it exists we get the rubric with the obtained id
+          const url = '/api/sites/' + this.siteId + '/rubrics/' + association.rubricId;
+          const resp = fetch(url, {
+            credentials: "include",
+            headers: { "Content-Type": "application/json" },
+          })
+          .then(r => {
+            if (r.ok) {
+              return r.json();
+            }
+
+            throw new Error("Network error while getting rubric");
+        })
+        .then(rubric => {
+          this.rubric = rubric;
+          if (rubric.criteria.length > 0) {
+            this.criterions = [];
+          } else {
+            this.criterions[0].description = this.i18n.criterion_default;
+          }
+          rubric.criteria.forEach(cr => {
+            let cNew =  {
+              pointsVal: Number(cr.ratings[0].points),
+              description: cr.title,
+              storedId: cr.id,
+              storedRatingId: cr.ratings[0].id,        
+            };
+            this.addRow(cNew);
+          })
+          this.storedCrit = this.processData();
+    
+          //get the evaluation
+          let url = `/api/sites/${this.siteId}/rubric-evaluations/tools/sakai.samigo/items/${this.entityId}/evaluations/${this.gradingId}/owners/${this.evaluatedItemOwnerId}`;
+          fetch(url, {
+            credentials: "include"
+          }).then(r => {
+            if (r.ok) {
+              return r.json();
+            }
+        // avoid console error first time?
+        if (r.status !== 404) {
+          throw new Error("Network error while getting evaluation");
+        }
+          }).then(evaluation => {
+            this.evaluation = evaluation || {
+              criterionOutcomes: []
+            };
+            this.storedEvaluation = this.evaluation;
+ 
+            this.decorateCriteria();
+	        this.updatePartial();
+          }).catch(error => console.error(error));
+
+          // link functions with samigo events
+          let $vm = this;
+          const host = this.$refs.component.getRootNode()?.host;
+          host.cancel = function() {
+            $vm.cancelChanges();
+          }
+          host.release = function() {
+            $vm.releaseChanges();
+          }
+
+        })
+        .catch (error => console.error(error));
+      });
+    },
+    cancelChanges() {
+      // reload original evaluation
+      this.evaluation = this.storedEvaluation;
+      this.decorateCriteria();
+    },
+    decorateCriteria() {
```

Comment:
I feel like this should be a computed property, keeping the original array intact and just using the decorated one. what do you think?

Reply:
This is actually just called when loading and then when reseting.

It makes a calculation to get difference between Samigo's grade and the selected criterions, then calls the toggleColor to select them if necessary. That method you mentioned it could be computed and I'll have a look (changing these calls if so), but this one feels a different case?

Reply:
Now I see it's modifying two state variables, so it might be better as a method actually. But maybe it would be useful to use a watcher on `this.evaluation` to call it, instead of doing it manually. But as you said, it's called in just two places, so maybe it's not worth it.